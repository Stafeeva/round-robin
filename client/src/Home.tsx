import React, { FC } from "react";
import { Button } from "antd";
import { Flex } from "antd";
import { Link } from "react-router-dom";

const Home: FC = () => {
  return (
    <Flex
      style={{ width: "100%", height: "100vh" }}
      align="center"
      vertical
      justify="space-around"
    >
      <Flex
        vertical
        gap="middle"
        style={{
          width: "400px",
          border: "1px solid #1677ff",
          borderRadius: "8px",
          padding: "72px",
        }}
      >
        <Button type="primary">
          <Link to="/create">Create a new meeting</Link>
        </Button>
        <Button type="primary">
          <Link to="/join">Join a meeting</Link>
        </Button>
      </Flex>
    </Flex>
  );
};

export default Home;
