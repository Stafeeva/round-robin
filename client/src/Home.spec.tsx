import React from "react";
import { render } from "@testing-library/react";

import Home from "./Home";

test("Home component renders", () => {
  const { container } = render(<Home />);
  expect(container).toMatchSnapshot();
});

test("User can choose to create a meeting", () => {
  const { getByText } = render(<Home />);
  const createMeetingButton = getByText("Create a new meeting");
  expect(createMeetingButton).toBeInTheDocument();
});

test("User can choose to join a meeting", () => {
  const { getByText } = render(<Home />);
  const joinMeetingButton = getByText("Join a new meeting");
  expect(joinMeetingButton).toBeInTheDocument();
});
