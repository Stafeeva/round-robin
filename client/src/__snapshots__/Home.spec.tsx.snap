// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders 1`] = `
<div>
  <div
    class="ant-flex css-dev-only-do-not-override-1uq9j6g ant-flex-align-center ant-flex-justify-space-around ant-flex-vertical"
    style="width: 100%; height: 100vh;"
  >
    <div
      class="ant-flex css-dev-only-do-not-override-1uq9j6g ant-flex-align-stretch ant-flex-gap-middle ant-flex-vertical"
      style="width: 400px; border: 1px solid #1677ff; border-radius: 8px; padding: 72px;"
    >
      <button
        class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-primary"
        type="button"
      >
        <span>
          Create a new meeting
        </span>
      </button>
      <button
        class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-primary"
        type="button"
      >
        <span>
          Join a new meeting
        </span>
      </button>
    </div>
  </div>
</div>
`;
